src/ts/services/keyring-vault.ts

```diff
@@ -0,0 +1,171 @@
+import { KeyringInstance, KeyringPair, KeyringPair$Json } from '@polkadot/keyring/types'
+import Keyring from '@polkadot/keyring'
+import { Prefix } from '@polkadot/keyring/address/types'
+import { LocalStore } from './local-store'
+import { mnemonicGenerate, cryptoWaitReady, mnemonicValidate } from '@polkadot/util-crypto'
+import t from './i18n'
+
+class KeyringVault {
+
+  private _vaultKey: string = 'speckle-vault'
+  private _keyring?: KeyringInstance
+  private _password?: string
+  private _mnemonic?: string
+
+  private get keyring (): KeyringInstance {
+    if (this._keyring) {
+      return this._keyring
+    }
+    throw new Error(`Keyring is not initialised yet`)
+  }
+
+  unlock (password: string, addressPrefix?: Prefix): Promise<Array<KeyringPair$Json>> {
+    if (this.isUnlocked()) throw new Error(t('error.wallet.unlocked'))
```

Comment:
Do we want to want to just return here if it is `isUnlocked`

Reply:
point taken